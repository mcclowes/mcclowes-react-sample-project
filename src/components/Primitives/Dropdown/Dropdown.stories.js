import Dropdown from "./DropdownContainer";
import React from "react";

export default { title: "Dropdown" };

export const empty = () => {
  return <Dropdown />;
};

export const withOptions = () => {
  return <Dropdown placeholder="Select..." options={[1, 2, 3, 4, 5]} />;
};

export const withCallback = () => {
  return (
    <Dropdown
      placeholder="Select..."
      options={[1, 2, 3, 4, 5]}
      callback={(value) => alert("You clicked " + value)}
    />
  );
};
