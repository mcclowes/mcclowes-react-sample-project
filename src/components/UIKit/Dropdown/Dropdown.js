import Icon from "../Icon";
import PropTypes from "prop-types";
import React from "react";

import {
  DropdownWrapper,
  DropdownPlaceholder,
  DropdownOptions,
  DropdownOption,
} from "./csx";

const Dropdown = (props) => {
  const { placeholder, options, open, setOpen, value, handleClick } = props;

  return (
    <DropdownWrapper onClick={() => setOpen(!open)}>
      <DropdownPlaceholder value={value} open={open}>
        {value === undefined ? placeholder : value}

        <Icon icon="DOWN" />
      </DropdownPlaceholder>

      {options && (
        <DropdownOptions open={open}>
          {options.map((option, i) => {
            return (
              <DropdownOption
                key={i}
                onClick={() => handleClick(option)}
                selected={value === option}
              >
                {option}
              </DropdownOption>
            );
          })}
        </DropdownOptions>
      )}
    </DropdownWrapper>
  );
};

Dropdown.propTypes = {
  handleClick: PropTypes.func,
  open: PropTypes.bool,
  options: PropTypes.array,
  placeholder: PropTypes.string,
  setOpen: PropTypes.func,
  value: PropTypes.any,
};

export default Dropdown;
