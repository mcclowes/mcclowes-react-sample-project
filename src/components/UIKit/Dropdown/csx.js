import styled from "styled-components/macro";

export const DropdownPlaceholder = styled.div`
  padding: 0.25em 0.5em;
  display: flex;
  justify-content: space-between;
  flex-direction: row;
  align-items: center;

  ${(props) =>
    props.value === undefined ? `color: ${props.theme.colors.gray};` : ""};
`;

DropdownPlaceholder.defaultProps = {
  theme: {
    colors: {
      gray: "#aaaaaa",
    },
  },
};

export const DropdownOptions = styled.div`
  ${(props) =>
    props.open
      ? `border-top: 1px solid ${props.theme.colors.gray};`
      : "display:none;"};
`;

DropdownOptions.defaultProps = {
  theme: {
    colors: {
      gray: "#aaaaaa",
    },
  },
};

export const DropdownOption = styled.div``;

export const DropdownWrapper = styled.div`
  border: 1px solid ${(props) => props.theme.colors.gray};
  border-radius: ${(props) => props.theme.sizes.border.radius};
  cursor: pointer;
  background: ${(props) => props.theme.colors.white};

  ${DropdownPlaceholder} {
    svg {
      fill: ${(props) => props.theme.colors.gray};
      transform: ${(props) => (props.open ? "rotate(180deg);" : "")};
    }
  }

  ${DropdownOption} {
    padding: 0.25em 0.5em;

    &:hover {
      background-color: ${(props) => props.theme.colors.lightgray};
    }

    &:active {
      background-color: ${(props) => props.theme.colors.gray};
    }
  }
`;

DropdownWrapper.defaultProps = {
  theme: {
    colors: {
      white: "#ffffff",
      gray: "#aaaaaa",
      lightgray: "#eeeeee",
    },
    sizes: {
      border: {
        radius: "3px",
      },
    },
  },
};
