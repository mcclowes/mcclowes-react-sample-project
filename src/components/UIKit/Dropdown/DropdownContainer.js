import Dropdown from "./Dropdown";
import PropTypes from "prop-types";
import React, { useState } from "react";

const DropdownContainer = (props) => {
  const [open, setOpen] = useState(false);
  const [value, setValue] = useState(undefined);

  const { callback } = props;

  const handleClick = (value) => {
    setValue(value);
    if (callback) {
      callback(value);
    }
  };

  return (
    <Dropdown
      value={value}
      handleClick={handleClick}
      open={open}
      setOpen={setOpen}
      {...props}
    />
  );
};

DropdownContainer.propTypes = {
  callback: PropTypes.func,
};

export default DropdownContainer;
