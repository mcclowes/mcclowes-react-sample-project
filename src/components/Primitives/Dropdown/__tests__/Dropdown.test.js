import Dropdown from "../DropdownContainer";
import React from "react";
import { mount } from "enzyme";

describe("Dropdown", () => {
  it("interaction", () => {
    const wrapper = mount(<Dropdown placeholder="Default Dropdown" />);

    expect(wrapper.find("csx__DropdownPlaceholder").prop("open")).toBe(false);

    wrapper.simulate("click");

    expect(wrapper).toMatchSnapshot();

    expect(wrapper.find("csx__DropdownPlaceholder").prop("open")).toBe(true);
  });
});
