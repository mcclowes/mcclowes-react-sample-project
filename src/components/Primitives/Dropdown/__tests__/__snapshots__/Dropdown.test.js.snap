// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown interaction 1`] = `
<DropdownContainer
  placeholder="Default Dropdown"
>
  <Dropdown
    handleClick={[Function]}
    open={true}
    placeholder="Default Dropdown"
    setOpen={[Function]}
  >
    <csx__DropdownWrapper
      onClick={[Function]}
      theme={
        Object {
          "colors": Object {
            "gray": "#aaaaaa",
            "lightgray": "#eeeeee",
            "white": "#ffffff",
          },
          "sizes": Object {
            "border": Object {
              "radius": "3px",
            },
          },
        }
      }
    >
      <StyledComponent
        forwardedComponent={
          Object {
            "$$typeof": Symbol(react.forward_ref),
            "_foldedDefaultProps": Object {
              "theme": Object {
                "colors": Object {
                  "gray": "#aaaaaa",
                  "lightgray": "#eeeeee",
                  "white": "#ffffff",
                },
                "sizes": Object {
                  "border": Object {
                    "radius": "3px",
                  },
                },
              },
            },
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "csx__DropdownWrapper-sc-1kkykl1-3",
              "isStatic": false,
              "lastClassName": "gYCuRG",
              "rules": Array [
                "border:1px solid ",
                [Function],
                ";border-radius:",
                [Function],
                ";cursor:pointer;background:",
                [Function],
                ";",
                ".csx__DropdownPlaceholder-sc-1kkykl1-0",
                "{svg{fill:",
                [Function],
                ";transform:",
                [Function],
                ";}}",
                ".csx__DropdownOption-sc-1kkykl1-2",
                "{padding:0.25em 0.5em;&:hover{background-color:",
                [Function],
                ";}&:active{background-color:",
                [Function],
                ";}}",
              ],
            },
            "displayName": "csx__DropdownWrapper",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "csx__DropdownWrapper-sc-1kkykl1-3",
            "target": "div",
            "toString": [Function],
            "usesTheme": true,
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          }
        }
        forwardedRef={null}
        onClick={[Function]}
        theme={
          Object {
            "colors": Object {
              "gray": "#aaaaaa",
              "lightgray": "#eeeeee",
              "white": "#ffffff",
            },
            "sizes": Object {
              "border": Object {
                "radius": "3px",
              },
            },
          }
        }
      >
        <div
          className="csx__DropdownWrapper-sc-1kkykl1-3 gYCuRG"
          onClick={[Function]}
        >
          <csx__DropdownPlaceholder
            open={true}
            theme={
              Object {
                "colors": Object {
                  "gray": "#aaaaaa",
                },
              }
            }
          >
            <StyledComponent
              forwardedComponent={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "_foldedDefaultProps": Object {
                    "theme": Object {
                      "colors": Object {
                        "gray": "#aaaaaa",
                      },
                    },
                  },
                  "attrs": Array [],
                  "componentStyle": ComponentStyle {
                    "componentId": "csx__DropdownPlaceholder-sc-1kkykl1-0",
                    "isStatic": false,
                    "lastClassName": "uhhfR",
                    "rules": Array [
                      "padding:0.25em 0.5em;display:flex;justify-content:space-between;flex-direction:row;align-items:center;",
                      [Function],
                      ";",
                    ],
                  },
                  "displayName": "csx__DropdownPlaceholder",
                  "foldedComponentIds": Array [],
                  "render": [Function],
                  "styledComponentId": "csx__DropdownPlaceholder-sc-1kkykl1-0",
                  "target": "div",
                  "toString": [Function],
                  "usesTheme": true,
                  "warnTooManyClasses": [Function],
                  "withComponent": [Function],
                }
              }
              forwardedRef={null}
              open={true}
              theme={
                Object {
                  "colors": Object {
                    "gray": "#aaaaaa",
                  },
                }
              }
            >
              <div
                className="csx__DropdownPlaceholder-sc-1kkykl1-0 uhhfR"
                open={true}
              >
                Default Dropdown
                <Icon
                  icon="DOWN"
                >
                  <csx__IconWrapper
                    focusable="false"
                    theme={
                      Object {
                        "colors": Object {
                          "black": "#333",
                        },
                      }
                    }
                    viewBox="0 0 32 32"
                  >
                    <StyledComponent
                      focusable="false"
                      forwardedComponent={
                        Object {
                          "$$typeof": Symbol(react.forward_ref),
                          "_foldedDefaultProps": Object {
                            "theme": Object {
                              "colors": Object {
                                "black": "#333",
                              },
                            },
                          },
                          "attrs": Array [],
                          "componentStyle": ComponentStyle {
                            "componentId": "csx__IconWrapper-sc-1flf2hb-0",
                            "isStatic": false,
                            "lastClassName": "fjmtxj",
                            "rules": Array [
                              "fill:",
                              [Function],
                              ";width:",
                              [Function],
                              ";height:",
                              [Function],
                              ";",
                            ],
                          },
                          "displayName": "csx__IconWrapper",
                          "foldedComponentIds": Array [],
                          "render": [Function],
                          "styledComponentId": "csx__IconWrapper-sc-1flf2hb-0",
                          "target": "svg",
                          "toString": [Function],
                          "usesTheme": true,
                          "warnTooManyClasses": [Function],
                          "withComponent": [Function],
                        }
                      }
                      forwardedRef={null}
                      theme={
                        Object {
                          "colors": Object {
                            "black": "#333",
                          },
                        }
                      }
                      viewBox="0 0 32 32"
                    >
                      <svg
                        className="csx__IconWrapper-sc-1flf2hb-0 fjmtxj"
                        focusable="false"
                        viewBox="0 0 32 32"
                      >
                        <path
                          d="M3.75,6.13,16,18.38,28.25,6.13,32,9.87l-16,16L0,9.87Z"
                        />
                      </svg>
                    </StyledComponent>
                  </csx__IconWrapper>
                </Icon>
              </div>
            </StyledComponent>
          </csx__DropdownPlaceholder>
        </div>
      </StyledComponent>
    </csx__DropdownWrapper>
  </Dropdown>
</DropdownContainer>
`;
